### DISCUSSION TITLE
How to revert to stock firmware

### DISCUSSION THREAD

**Discussion by sirtow** [emporia-vue-local/esphome#246] [Q&A]
I want to test test original firmware but i cannot find a way to restore my firmware. It seems after flashing ESPHome , i cannot get the unit into bootloader mode.

Is it possible that after flashing ESPHome the bootloader mode is disabled? or is there any other sequence (besides pressing IO0 against GND) to start the bootloader?

Every time i try, i get `dev/ttyUSB0 failed to connect: Failed to connect to Espressif device: No serial data received.`

The unit still boots into ESPHome (when IO0 is not grounded)  and i see it connected to WIFI.

**Comment by flaviut** [comment]
> Is it possible that after flashing ESPHome the bootloader mode is disabled?

No, the bootloader on the esp32 cannot be disabled or altered.

Do you maybe have your rx and tx reversed? Or a loose connection?
